python/lsst/summit/extras/plotting/psfPlotting.py

```diff
@@ -0,0 +1,524 @@
+    """
+    tables = []
+
+    for detectorNum, icSrc in icSrcs.items():
+        icSrc = icSrc.asAstropy()
+        icSrc = icSrc[icSrc["calib_psf_candidate"]]
+        icSrc["detector"] = detectorNum
+        tables.append(icSrc)
+
+    table = vstack(tables)
+    # Add shape columns
+    table["Ixx"] = table["slot_Shape_xx"] * (0.2) ** 2
+    table["Ixy"] = table["slot_Shape_xy"] * (0.2) ** 2
+    table["Iyy"] = table["slot_Shape_yy"] * (0.2) ** 2
+    table["T"] = table["Ixx"] + table["Iyy"]
+    table["e1"] = (table["Ixx"] - table["Iyy"]) / table["T"]
+    table["e2"] = 2 * table["Ixy"] / table["T"]
+    table["e"] = np.hypot(table["e1"], table["e2"])
+    table["x"] = table["base_FPPosition_x"]
+    table["y"] = table["base_FPPosition_y"]
+
+    table.meta["rotTelPos"] = (
+        visitInfo.boresightParAngle - visitInfo.boresightRotAngle - (np.pi / 2 * radians)
+    ).asRadians()
+    table.meta["rotSkyPos"] = visitInfo.boresightRotAngle.asRadians()
+
+    rtp = table.meta["rotTelPos"]
+    srtp, crtp = np.sin(rtp), np.cos(rtp)
+    aaRot = np.array([[crtp, srtp], [-srtp, crtp]]) @ np.array([[0, 1], [1, 0]]) @ np.array([[-1, 0], [0, 1]])
+    table = extendTable(table, aaRot, "aa")
+    table.meta["aaRot"] = aaRot
+
+    rsp = table.meta["rotSkyPos"]
+    srsp, crsp = np.sin(rsp), np.cos(rsp)
+    nwRot = np.array([[crsp, -srsp], [srsp, crsp]])
+    table = extendTable(table, nwRot, "nw")
+    table.meta["nwRot"] = nwRot
+
+    return table
+
+
+def extendTable(table: Table, rot: npt.NDArray[np.float_], prefix: str) -> Table:
+    """Extend the given table with additional columns for the rotated shapes.
+
+    Parameters
+    ----------
+    table : `astropy.table.Table`
+        The input table containing the original shapes.
+    rot : `np.ndarray`
+        The rotation matrix used to rotate the shapes.
+    prefix : `str`
+        The prefix to be added to the column names of the rotated shapes.
+
+    Returns
+    -------
+    table : `astropy.table.Table`
+        The extended table with additional columns representing the rotated
+        shapes.
+    """
+    transform = LinearTransform(rot)
+    rot_shapes = []
+    for row in table:
+        shape = Quadrupole(row["Ixx"], row["Iyy"], row["Ixy"])
+        rot_shape = shape.transform(transform)
+        rot_shapes.append(rot_shape)
+    table[prefix + "_Ixx"] = [sh.getIxx() for sh in rot_shapes]
+    table[prefix + "_Iyy"] = [sh.getIyy() for sh in rot_shapes]
+    table[prefix + "_Ixy"] = [sh.getIxy() for sh in rot_shapes]
+    table[prefix + "_e1"] = (table[prefix + "_Ixx"] - table[prefix + "_Iyy"]) / table["T"]
+    table[prefix + "_e2"] = 2 * table[prefix + "_Ixy"] / table["T"]
+    table[prefix + "_x"] = rot[0, 0] * table["x"] + rot[0, 1] * table["y"]
+    table[prefix + "_y"] = rot[1, 0] * table["x"] + rot[1, 1] * table["y"]
+    return table
+
+
+def makeFigureAndAxes() -> tuple[plt.Figure, np.ndarray[plt.Axes]]:
```

Comment:
If MyPy isn't complaining about the second half of the return type for you, it's probably seeing `np.ndarray` as `Any`, since that usually requires two type parameters.  (FWIW, I dislike Numpy's typing enough that I configure MyPy to treat `np.ndarray` as `Any` and then just use bare `np.ndarray`, but if you're going to add type parameters, it should be correct.)

Reply:
Thanks, just changing to `Any` for now because matplotlib + mypy honestly makes me lose the will to live.